106_A2_M1/Model/Singleton.cs

```diff
@@ -95,15 +117,271 @@ public async Task<int> LoginAsync(string email, string password, string authToke
                 return 0;
             }
         }
+        private async Task<UserDB> GetUserInformationAsync()
+        {
+            try
+            {
+                // Make a request to your backend API to retrieve user information
+                HttpResponseMessage userResponse = await this._client.GetAsync("https://cse106-backend.d3rpp.dev/api/user");
+
+                if (userResponse.IsSuccessStatusCode)
+                {
+                    // Deserialize the response content to obtain the user information
+                    string content = await userResponse.Content.ReadAsStringAsync();
+                    var userData = JsonConvert.DeserializeObject<UserDataFromBackend>(content);
+
+                    // Map the backend data to UserDB class
+                    UserDB userDB = new UserDB
+                    {
+                        email = userData.email,
+                        first_name = userData.given_name,
+                        last_name = userData.family_name,
+                        dob = userData.dob_ts,
+                        id = userData.id,
+                        qr_status = userData.qrcode_status ?? 0, // Handle possible null value
+                        issue_ct = userData.issue_count,
+                        test_ct = userData.test_count,
+                        vaccine_status = userData.vaccine_status
+                    };
+
+                    return userDB;
+                }
+                else
+                {
+                    Console.WriteLine($"Error fetching user information: {userResponse.StatusCode} - {userResponse.ReasonPhrase}");
+                    return null;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"An error occurred while fetching user information: {ex.Message}");
+                return null;
+            }
+        }
+
+        // Class to represent the data received from the backend
+        public class UserDataFromBackend
```

Comment:
I've already written an Item for this that makes this irrelevant which also comes with a function to get it from a JSON string.

See https://github.com/SpursFan21/106_A2_M1/blob/master/106_A2_M1/Interfaces/User/User.cs